### PR TITLE
feat: rag fusion notebook

### PR DESCRIPTION
**Pull request by bracesproul**

### CODE REVIEW

File: cookbook/rag_fusion.ipynb [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+    "A LangChain JS port of (this Github repo)[https://github.com/Raudaschl/rag-fusion], all credit to the original author.\n",
+    "> RAG-Fusion, a search methodology that aims to bridge the gap between traditional search paradigms and the multifaceted dimensions of human queries. Inspired by the capabilities of Retrieval Augmented Generation (RAG), this project goes a step further by employing multiple query generation and Reciprocal Rank Fusion to re-rank search results."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup\n",
+    "For this example we'll use HSNW as our vector store/retriever, and some fake data.\n"
```
Comment (jacoblee93) [inline cookbook/rag_fusion.ipynb:17]: Not HNSW anymore?

File: cookbook/rag_fusion.ipynb [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 69,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "const chain = RunnableSequence.from([\n",
+    "  generateQueries,\n",
+    "  retriever.map(),\n",
```
Comment (jacoblee93) [inline cookbook/rag_fusion.ipynb:199]: Ah I forgot this method existed 😅

Comment (jacoblee93) [reply cookbook/rag_fusion.ipynb:199]: Can you make a note to document in the LCEL section?

**Review by jacoblee93 — CHANGES_REQUESTED** [review]
Couple of small fixes but looks good - can you also add it to the main `cookbook/README.md` file?

Also might be good to push a live version to Noteable

File: cookbook/rag_fusion.ipynb (hunk 2 above)
Comment (bracesproul) [reply cookbook/rag_fusion.ipynb:199]: tracking here https://github.com/langchain-ai/langchainjs/issues/3052

**Review by jacoblee93 — APPROVED** [review]

**Comment by rgbkrk** [comment]
It looks like the published version at https://app.noteable.io/published/d9902d51-c5e9-4d89-bcb1-f82521ab4497/rag_fusion is missing the code bits. Maybe a new version needs to be published or just make the original notebook be public.

**Comment by jacoblee93** [comment]
CC @bracesproul!

**Pull request merged**